tests/unit/core/sagas/test_user.js

```diff
@@ -0,0 +1,63 @@
+import SagaTester from 'redux-saga-tester';
+
+import userSaga from 'core/sagas/user';
+import userReducer, { userProfileLoaded } from 'core/reducers/user';
+import * as api from 'core/api/user';
+import apiReducer from 'core/reducers/api';
+import { setAuthToken } from 'core/actions';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import { createUserProfileResponse, userAuthToken } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let mockApi;
+  let sagaTester;
+  let rootTask;
+
+  beforeEach(() => {
+    mockApi = sinon.mock(api);
+    const initialState = dispatchSignInActions().state;
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: {
+        api: apiReducer,
+        user: userReducer,
+      },
+    });
+    rootTask = sagaTester.start(userSaga);
+  });
+
+  it('calls the API to fetch user profile after setAuthToken()', async () => {
+    const profile = createUserProfileResponse();
+
+    mockApi
+      .expects('userProfile')
+      .once()
+      .returns(Promise.resolve(profile));
+
+    sagaTester.dispatch(setAuthToken(userAuthToken()));
+
+    const expectedCalledAction = userProfileLoaded({ profile });
+
+    await sagaTester.waitFor(expectedCalledAction.type);
+    mockApi.verify();
+
+    const calledAction = sagaTester.getCalledActions()[1];
+    expect(calledAction).toEqual(expectedCalledAction);
+  });
+
+  it('lets exceptions to be thrown', () => {
```

Comment:
I think this test spec could be better written as `it('handles exceptions and dispatches an error', () => { [...] })`

But then as mentioned before I think the way this saga handles errors isn't how we tend to. I guess we should write some documentation at least on error handling, because we have a HOC and reducer for dispatching errors that components can consume.